en/03_Drawing_a_triangle/00_Setup/02_Validation_layers.adoc

```diff
@@ -117,96 +130,113 @@ void createInstance() {
+`VK_EXT_DEBUG_UTILS_EXTENSION_NAME` macro here which is equal to the literal
+string "VK_EXT_debug_utils". Using this macro lets you avoid typos.
 
 We can now use this function in `createInstance`:
 
 [,c++]
 ----
 auto extensions = getRequiredExtensions();
-createInfo.enabledExtensionCount = static_cast<uint32_t>(extensions.size());
-createInfo.ppEnabledExtensionNames = extensions.data();
+vk::InstanceCreateInfo createInfo({}, &appInfo, enabledLayers, extensions);
 ----
 
-Run the program to make sure you don't receive a `VK_ERROR_EXTENSION_NOT_PRESENT` error.
-We don't really need to check for the existence of this extension, because it should be implied by the availability of the validation layers.
+Run the program to make sure you don't receive a
+`VK_ERROR_EXTENSION_NOT_PRESENT` error. We don't really need to check for the
+existence of this extension because it should be implied by the availability of
+the validation layers.
 
-Now let's see what a debug callback function looks like.
-Add a new static member function called `debugCallback` with the `PFN_vkDebugUtilsMessengerCallbackEXT` prototype.
-The `VKAPI_ATTR` and `VKAPI_CALL` ensure that the function has the right signature for Vulkan to call it.
+Now let's see what a debug callback function looks like. Add a new static member
+function called `debugCallback` with the `PFN_vkDebugUtilsMessengerCallbackEXT`
+prototype. The `VKAPI_ATTR` and `VKAPI_CALL` ensure that the function has the
+right signature for Vulkan to call it.
 
 [,c++]
 ----
-static VKAPI_ATTR VkBool32 VKAPI_CALL debugCallback(
-    VkDebugUtilsMessageSeverityFlagBitsEXT messageSeverity,
-    VkDebugUtilsMessageTypeFlagsEXT messageType,
-    const VkDebugUtilsMessengerCallbackDataEXT* pCallbackData,
-    void* pUserData) {
-
-    std::cerr << "validation layer: " << pCallbackData->pMessage << std::endl;
+static VKAPI_ATTR vk::Bool32 VKAPI_CALL debugCallback(vk::DebugUtilsMessageSeverityFlagBitsEXT severity, vk::DebugUtilsMessageTypeFlagsEXT type, const vk::DebugUtilsMessengerCallbackDataEXT* pCallbackData, void*) {
+    std::cerr << "validation layer: type " << to_string(type) << " msg: " << pCallbackData->pMessage << std::endl;
 
-    return VK_FALSE;
+    return vk::False;
 }
 ----
 
-The first parameter specifies the severity of the message, which is one of the following flags:
+The first parameter specifies the severity of the message, which is one of
+the following flags:
 
 * `VK_DEBUG_UTILS_MESSAGE_SEVERITY_VERBOSE_BIT_EXT`: Diagnostic message
-* `VK_DEBUG_UTILS_MESSAGE_SEVERITY_INFO_BIT_EXT`: Informational message like the creation of a resource
-* `VK_DEBUG_UTILS_MESSAGE_SEVERITY_WARNING_BIT_EXT`: Message about behavior that is not necessarily an error, but very likely a bug in your application
-* `VK_DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT`: Message about behavior that is invalid and may cause crashes
+* `VK_DEBUG_UTILS_MESSAGE_SEVERITY_INFO_BIT_EXT`: Informational message
```

Comment:
Should be updated to use the vulkan.hpp enums instead (e.g. vk::DebugUtilsMessageSeverityFlagBitsEXT::eWarning)

Reply:
I left it with the C version in the tutorial text on purpose with the idea that we can link directly to the spec such that the reader will know to look for the spec's version instead of the C++ version.  That might be confusing though, maybe we should change it to C++ everywhere?